packages/rx/src/Result.ts

```diff
@@ -314,3 +314,31 @@ export const map: {
       return success(f(self.value), self.waiting)
   }
 })
+
+/**
+ * @since 1.0.0
+ * @category types
+ */
+export interface MatchOptions<A, E, Ret> {
+  onInitial: (_: Initial<A, E>) => Ret
+  onFailure: (_: Failure<A, E>) => Ret
+  onSuccess: (_: Success<A, E>) => Ret
+}
+
+/**
+ * @since 1.0.0
+ * @category combinators
+ */
+export const match: {
+  <A, E, Ret>(options: MatchOptions<A, E, Ret>): (self: Result<A, E>) => Ret
```

Comment:
I think inlining the options is ok.

```ts
options: {
  readonly onInitial: (_: Initial<A, E>) => X
  readonly onFailure: (_: Failure<A, E>) => Y
  readonly onSuccess: (_: Success<A, E>) => Z
}
```

Reply:
Similar to how .match is usually implemented in the `effect` modules: https://effect-ts.github.io/effect/effect/Either.ts.html#match

Reply:
It just means the type signature is more obvious at a glance, instead of having to jump to the interface to see all the options.

Reply:
Ah gotcha. Makes sense. Updated :)